crates/transaction-emitter-lib/src/transaction_generator/nft_mint_and_transfer.rs

```diff
@@ -148,32 +180,34 @@ pub fn create_nft_token_request(
 }
 
 pub fn create_nft_transfer_request(
-    owner_account: &mut LocalAccount,
-    creation_account: &LocalAccount,
+    sender: &mut LocalAccount,
+    receiver: &LocalAccount,
+    creation_address: AccountAddress,
     collection_name: &[u8],
     token_name: &[u8],
     txn_factory: &TransactionFactory,
+    amount: u64,
 ) -> SignedTransaction {
-    owner_account.sign_multi_agent_with_transaction_builder(
-        vec![creation_account],
+    sender.sign_multi_agent_with_transaction_builder(
+        vec![receiver],
         txn_factory.payload(aptos_token_stdlib::token_direct_transfer_script(
-            creation_account.address(),
+            creation_address,
             collection_name.to_vec(),
             token_name.to_vec(),
             0,
-            1,
+            amount,
         )),
     )
 }
 
-pub struct NFTMintGeneratorCreator {
+pub struct NFTMintAndTransferGeneratorCreator {
     txn_factory: TransactionFactory,
-    creator_account: Arc<LocalAccount>,
+    creator_account: Arc<Mutex<LocalAccount>>,
```

Comment:
This is a root account?

I think we shouldn't use a root_account for anything beyond initialization (i.e. initialize_nft_collection), as that gets used in other part of the system, and can get out of sync.

so can we have at the beginning root_account create a creator_account which can then mint nfts (and is fully owned by this class)? (I assume root account doesn't need to mint NFTs to the creator account?)

Reply:
Root account is a separate account. This creator account is sort of a faucet account that only for nft minting in fact.

Reply:
ah I missed the line
let mut creator_account = LocalAccount::generate(&mut rng);
below.